src/Reduction.h

```diff
@@ -46,6 +46,27 @@ class ReductionDomain {
 
     /** Immutable access to the reduction variables. */
     EXPORT const std::vector<ReductionVariable> &domain() const;
+
+    /** Add predicate to the reduction domain. See \ref RDom::where
+     * for more details. */
+    EXPORT void where(const Expr &predicate);
+
+    /** Get the predicates defined on this reducation demain if there is any. */
+    EXPORT const std::vector<Expr> &predicates() const;
+    EXPORT std::vector<Expr> &predicates();
```

Comment:
Why do you need mutable access to the predicates vector? Might it be preferable to have an add_predicate method?